### PR TITLE
chore: jest config normalization

### PR DESCRIPTION
**Pull request by sdstolworthy**
#### Description of changes

This PR moves all jest configurations to the `"jest"` block of each project's `package.json`. Additionally, a `collectCoverageFrom` option is added to each Jest config.

In some projects, Jest configs were stored in both a `jest.config.js` file and the `package.json` `"jest"` block. This is a no-op. Since the majority of projects were using the Jest config in `package.json`, all packages were standardized on that norm.

Additionally, a `collectCoverageFrom` option was added to each Jest config file. In the absence of the `collectCoverageFrom` option, coverage statistics were only collected on files that are actually loaded during testing. This can lead to underreporting coverage.

See [the jest documentation](https://jestjs.io/docs/cli#--collectcoveragefromglob).
#### Description of how you validated changes

Ran tests, verified tests passing. Verified coverage was collected from a sampling of previously missing files.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by lazpavel — DISMISSED** [review]

**Review by edwardfoyle — DISMISSED** [review]

File: packages/amplify-dynamodb-simulator/package.json [hunk 1]
```diff
@@ -40,6 +40,10 @@
   },
   "jest": {
     "testEnvironment": "node",
-    "collectCoverage": true
+    "collectCoverage": true,
+    "collectCoverageFrom": [
+      "src/**/*.{ts,tsx,js,jsx}",
+      "!src/__tests__/"
+    ]
```
Comment (jhockett) [inline packages/amplify-dynamodb-simulator/package.json:47]: This package has a non-standard folder structure (no src folder), so this needs to be modified

Comment (sdstolworthy) [reply packages/amplify-dynamodb-simulator/package.json:47]: Done, good catch

File: packages/amplify-frontend-ios/package.json [hunk 2]
```diff
@@ -29,6 +29,10 @@
   },
   "jest": {
     "collectCoverage": true,
+    "collectCoverageFrom": [
```
Comment (jhockett) [inline packages/amplify-frontend-ios/package.json:32]: Sorry didn't think to check earlier, but most of the FE packages have source code in `lib` and the package root

**Review by lazpavel — APPROVED** [review]
Please trigger e2e tests

**Review by evcodes — APPROVED** [review]
LGTM!

**Review by jhockett — APPROVED** [review]

**Pull request merged**